src/System.Buffers.Experimental/System/Buffers/SpanWriter.cs

```diff
@@ -192,6 +257,42 @@ private void Resize()
             _buffer = newBuffer.Span;
         }
 
+        private bool TryApplyTransformation(int index, int length, IBufferTransformation transformation)
+        {
+            Debug.Assert(transformation != null);
+            var status = transformation.Transform(Buffer.Slice(index), length, out int transformed);
+            if (status == OperationStatus.Done)
+            {
+                _written += transformed;
+                return true;
+            }
+
+            if (status == OperationStatus.InvalidData || status == OperationStatus.NeedMoreData) throw new Exception("invalid value or transformation");
+            
+            return true;
+        }
+
+        private bool TryApplyTransformations(int index, int length, ReadOnlySpan<IBufferTransformation> transformations)
+        {
+            var buffer = Buffer.Slice(index);
+            for (int i = 0; i < transformations.Length; i++)
+            {
+                var status = transformations[i].Transform(buffer, length, out length);
```

Comment:
Use a differently named local for the out parameter instead of length.
What happens if transformations.Length == 0? We end up setting _written += length (length being what the user passed in). Is that expected?

Reply:
Yes, if the last transformation produced no bytes, then no bytes have been written. i.e. using length here is by design.